text/0000-optional-jquery.md

```diff
@@ -0,0 +1,248 @@
+exists  [ember-native-dom-helpers-codemod](https://github.com/simonihmig/ember-native-dom-helpers-codemod), which 
+could be adapted to the very similar RFC 268 based interaction helpers in `@ember/test-helpers`.
+
+### Implementation outline
+
+The following outlines how a possible implementation of the jQuery integration flag *could* look like. This
+is just to provide some additional context, but is *intentionally not* meant to be normative, to allow some flexibility 
+for the actual implementation.
+
+The addon that will handle the flag is expected to be [ember-optional-features](https://github.com/emberjs/ember-optional-features),
+which will read from and write to a `config/optional-features.{js,json}` file. This will hold the `jquery-integration` 
+flag (amongst others). This flag in turn will be added to the `EmberENV` hash, which will make Ember go into its 
+"no jQuery" mode when set to `false`. 
+
+Ember CLI and the `@ember/jquery` addon will also look for `jquery-integration` in this configuration file, and will 
+opt-out of importing jQuery when this file is present and the flag is set to `false`.
+
+## How we teach this
+
+### Guides
+
+The existing "Managing Dependencies" chapters in the Ember Guides as well as on ember-cli.com provide a good place to 
+explain users how to set the jQuery integration flag by means of the mentioned addon that handles this flag. 
+
+The section on components should be updated to remove any eventually remaining references to `this.$`, to not let users 
+fall into the trap of creating an implicit dependency on jQuery by "accidental" use of it. These should be changed to 
+refer to their native DOM counterparts like `this.element` or `this.element.querySelector()`.
+
+The section on acceptance tests will have been updated as per [RFC 268](https://github.com/emberjs/rfcs/blob/master/text/0268-acceptance-testing-refactor.md)
+to use the new `@ember/test-helpers` based test helpers instead of the jQuery based global helpers.
+
+The section on component tests should not use `this.$()` anymore as well, and instead also according to [RFC 268](https://github.com/emberjs/rfcs/blob/master/text/0268-acceptance-testing-refactor.md) 
+use `this.element` to refer to the component's root element, and use the new DOM interaction helpers instead of jQuery 
+events triggered through `this.$()`. 
+
+### Addon migration
+
+One of the biggest problems to easily opt-out of jQuery is that many addons still depend on it. Many of these usages 
+seem to be rather "accidental", in that the full power of jQuery is not really needed for the given task, and could 
+be fairly easily refactored to use only native DOM APIs.
+
+For this reason this RFC encourages addon authors to not use jQuery anymore and to refactor existing usages whenever 
+possible! This certainly does not apply categorically to all addons, e.g. those that wrap jQuery plugins as 
+components and as such cannot drop this dependency.
```

Comment:
Addons that require jquery should state that they do so.

At some point the end goal should be that such addons could simply include `@ember/jquery` in their dependencies and not have additional instructions.

Reply:
> Addons that require jquery should state that they do so.

Totally agreed! The current line of thinking is that addons that require jQuery (or any of the other optional features that we are adding opt-out capabilities for) can as `@ember/optional-features` if any given feature is enabled or disabled, this allows an addon that is (for example) wrapping a jQuery plugin ensure that jQuery integration is enabled (and `@ember/jquery` is present properly). 

> At some point the end goal should be that such addons could simply include @ember/jquery in their dependencies and not have additional instructions.

_Possibly_, but for now I think that we want to leave control of opting in or out of features in the hands of the application specifically.